[BUG] ReorderGroup is incompatible with React 19 types

**1. Read the [FAQs](#faqs) 👇**

**2. Describe the bug**

https://github.com/motiondivision/motion/pull/3368 updated the types of `ReorderGroupComponent` to return `JSX.Element`, but `JSX` is no longer a global namespace in the React 19 types. See https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript

This results in a type error when Motion is imported in a project using React 19:

```
node_modules/framer-motion/dist/types/index.d.ts:515:226 - error TS2503: Cannot find namespace 'JSX'.

515 declare function ReorderGroupComponent<V, TagName extends ReorderElementTag = DefaultGroupElement>({ children, as, axis, onReorder, values, ...props }: ReorderGroupProps<V, TagName>, externalRef?: React$1.ForwardedRef<any>): JSX.Element;
```

You could use `React.JSX.Element` but more likely you want `React.ReactNode` which is the return type for React components in general.

**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://github.com/billyjanitsch/motion-react-types-bug

**4. Steps to reproduce**

Run `npm install && npm test` in the minimal repo above.

**5. Expected behavior**

No type error.

**6. Video or screenshots**

n/a

**7. Environment details**

n/a

--- a/packages/framer-motion/src/components/Reorder/Item.tsx
+++ b/packages/framer-motion/src/components/Reorder/Item.tsx
@@ -65,7 +65,7 @@ export function ReorderItemComponent<
         ...props
     }: ReorderItemProps<V, TagName>,
     externalRef?: React.ForwardedRef<any>
-): JSX.Element {
+): React.JSX.Element {
     const Component = useConstant(
         () => motion[as as keyof typeof motion]
     ) as FunctionComponent<

